third_party/move/move-model/src/ast.rs

```diff
@@ -281,6 +281,8 @@ pub struct Spec {
     pub properties: PropertyBag,
     // If this is a function, specs associated with individual code points.
     pub on_impl: BTreeMap<CodeOffset, Spec>,
+    // The map to store ghost variable update statements inlined in the function body
+    pub update_map: BTreeMap<NodeId, Condition>,
```

Comment:
This seems to contain a map from the node id of inlined conditions in this function to the actual condition? I do not understand this construct. In order to get something out of this map, you need to have a `cond.node_id()`, and then you get `Some(cond')` back s.t. `cond == cond'`. Or am I missing something? Is this perhaps just better a `updates: BTreeSet<NodeId>`?

Reply:
This map is updated in [stackless_bytecode_generator.rs](https://github.com/aptos-labs/aptos-core/blob/2465c0799e1697df63b62f00411dec731f87b3fe/third_party/move/move-prover/bytecode/src/stackless_bytecode_generator.rs#L224), when we see an update of global spec variables stored in `cond`. Later, when we want to populate the field `updates` inside `translator.result` in the function `translate_inline_property`, we need to get [access](https://github.com/aptos-labs/aptos-core/blob/2465c0799e1697df63b62f00411dec731f87b3fe/third_party/move/move-model/src/spec_translator.rs#L286) to `cond`. With this map, we no longer need to [traverse](https://github.com/aptos-labs/aptos-core/blob/0a6bc636567e2e831f2a6f260bb961c36298201e/third_party/move/move-model/src/spec_translator.rs#L286) values stored in `translator.fun_env.get_spec().on_impl.values()` to get `cond`.

Reply:
Oh I see. Wow this was a terrible hack before ... (`cond.exp == prop.clone()`). Now it is slightly better, still somehow the abstractions are wrong. We are getting an exp and via its node id identify its parent condition. This is wild but a change would be a larger refactoring.

Reply:
Thanks for the comment, Wolfgang! We could discuss more on how to refactor it later

Reply:
Hmm, PR merged. NVM, next time then.